connectors/byoc.py

```diff
@@ -212,6 +212,38 @@ async def done(self, indexed_count=0, deleted_count=0, exception=None):
 
         return await self.client.update(index=JOBS_INDEX, id=self.job_id, doc=job_def)
 
+    @classmethod
+    def transform_filtering(cls, filtering):
+        # deepcopy to not change the reference resulting in changing .elastic-connectors filtering
+        filtering = (
+            {"advanced_snippet": {}, "rules": []}
+            if (filtering is None or len(filtering) == 0)
+            else deepcopy(filtering)
+        )
+
+        # extract value for sync job
+        filtering["advanced_snippet"] = filtering.get("advanced_snippet", {}).get(
+            "value", {}
+        )
```

Comment:
If I read it correctly, the structure of filtering in `.elastic-connectors` is
```
{
  "advanced_sinppet": {
    "value": "foobar"
  }
}
```

and you want to transform it to

```
{
  "advanced_sinppet": "foobar"
}
```

May I know why? the `connector` field of `.elastic-connectors-sync-jobs` should be just a copy of the fields of `.elastic-connectors`

Reply:
This was stated in such a way in the original implementation doc and it's implemented this way in Ruby (Ruby actually supports both cases; will add that to python, too) Ruby implementation: `'advanced_snippet' => snippet['value'] || snippet`. Wanted to keep both consistent to not break it for a different protocol version. I'll probably create a follow-up issue to use one (and the same) route for ruby and python.

Reply:
Hah, I see it in ruby